src/run.ts

```
@@ -52,18 +63,20 @@ export async function run() {
       )
       core.debug(`Writing kubeconfig to ${kubeconfigPath}`)
       const cmd = [
-         'aks',
+         resourceType.toLowerCase()=='microsoft.containerservice/fleets' ? 'fleet':'aks',
          'get-credentials',
          '--resource-group',
          resourceGroupName,
          '--name',
-         clusterName,
+         resourceName,
          '-f',
          kubeconfigPath
       ]
       if (subscription) cmd.push('--subscription', subscription)
-      if (admin) cmd.push('--admin')
-      if (publicFqdn) cmd.push('--public-fqdn')
+      if (resourceType!='fleet') {
+         if (admin) cmd.push('--admin')
```

ideally should probably validate above against user setting (admin || publicFqdn) && fleet

---

Reactivating.  What I mean is, up above by `// check resource type is recognized`, I think you should also validate:

```js
if (resourceType === 'microsoft.containerservice/fleets' && admin) {
    throw Error('admin must not be true when resource type is Microsoft.ContainerService/fleets')
}
if (resourceType === 'microsoft.containerservice/fleets' && publicFqdn) {
    throw Error('public-fqdn must not be true when resource type is Microsoft.ContainerService/fleets')
}
```

Then you don't need any changes in this part of the file.

This way, an invalid configuration is specifically rejected early, which helps the customer, rather than ignoring it, which could confuse them.